### ISSUE TITLE
[Bug]: Pie chart autopcr % symbol disappears with text.usetex="True"

### ISSUE THREAD

**Issue by pavaninguva** [matplotlib/matplotlib#26377]
### Bug summary

When I plot a simple bar chart and set `plt.rcParams["text.usetex"] = True`, the % symbol in the bar chart disappears. It makes sense considering that % is treated as a comment within Latex, but there seems to be no way to override this. When I comment out that line, things return to normal. 

### Code for reproduction

```python
import matplotlib.pyplot as plt

#formatting
plt.rcParams["text.usetex"] = True
plt.rc('font', family='serif')
plt.rc("font",size=11)

first_year = [20,10,70]
colors = ["deepskyblue","steelblue","darkblue"]
legend=["Coursework", "Practical","Examination"]

fig1,ax1 = plt.subplots(1,1)
ax1.pie(first_year,colors=colors,autopct="%1.0f%%",textprops={'color':"w","weight":"bold"},radius=2,wedgeprops=dict(width=1.2),pctdistance=0.65)
ax1.set_title("Year 1",y=1.5,weight='bold')
fig1.legend(legend,ncol=3,loc="lower center")
fig1.tight_layout()

plt.show()
```

### Actual outcome

![mwe_tex](https://github.com/matplotlib/matplotlib/assets/41192490/d394f4af-26b8-45ac-9ee3-27b5e99f589f)

### Expected outcome

It should look like this, except the text should be in the Latex font

![mwe_notex](https://github.com/matplotlib/matplotlib/assets/41192490/65d25a1b-f46b-43f0-855e-17b0be2c311b)

### Operating system

Ubuntu on WSL

### Matplotlib Version

3.7.1

### Matplotlib Backend

QtAgg

### Python version

3.10.10

### Installation

conda

**Comment by anntzer** [comment]
I suspect it would be nice to have something like 972b815c596792c0f40a3a368f44e5b8eada3b26 for usetex.

**Comment by Tikooh** [comment]
Hi, I'm quite new to contributing. I believe the issue lies in specifying the format string for autopct.

Instead of `autopct="%1.0f%%"`
Try using `autopct="%1.0f\%%"`

### Outcome

![image](https://github.com/matplotlib/matplotlib/assets/114261455/edb123c4-7a3f-4557-a722-22c747274289)

**Comment by Priyanshu72** [comment]
how to solve this problem , I did not get snippet in the repository

**Comment by Priyanshu72** [comment]
i am new in this field , I don't know how to find issues and solve that problem.
Please help.

**Comment by praetorian2710** [comment]
import matplotlib.pyplot as plt

# formatting
plt.rcParams["text.usetex"] = True
plt.rc('font', family='serif')
plt.rc("font", size=11)

first_year = [20, 10, 70]
colors = ["deepskyblue", "steelblue", "darkblue"]
legend = ["Coursework", "Practical", "Examination"]

fig1, ax1 = plt.subplots(1, 1)
ax1.pie(first_year, colors=colors, autopct="%1.0f\\%%", textprops={'color': "w", "weight": "bold"},
        radius=2, wedgeprops=dict(width=1.2), pctdistance=0.65)
ax1.set_title("Year 1", y=1.5, weight='bold')
fig1.legend(legend, ncol=3, loc="lower center")
fig1.tight_layout()

plt.show()


This works completely but how to contribute?

**Comment by praetorian2710** [comment]
`import matplotlib.pyplot as plt

plt.rcParams["text.usetex"] = True
plt.rc('font', family='serif')
plt.rc("font", size=11)

first_year = [20, 10, 70]
colors = ["deepskyblue", "steelblue", "darkblue"]
legend = ["Coursework", "Practical", "Examination"]

fig1, ax1 = plt.subplots(1, 1)
ax1.pie(first_year, colors=colors, autopct="%1.0f\\%%", textprops={'color': "w", "weight": "bold"},
        radius=2, wedgeprops=dict(width=1.2), pctdistance=0.65)
ax1.set_title("Year 1", y=1.5, weight='bold')
fig1.legend(legend, ncol=3, loc="lower center")
fig1.tight_layout()

plt.show()`



This works perfectly how to contribute?

**Comment by Tikooh** [comment]
Whenever I tried to make one it said "Pull request creation failed. Validation failed: must be a collaborator"

**Comment by melissawm** [comment]
@praetorian2710 @Priyanshu72 it looks like there are two different things here: one is that you can maybe bypass the escaping issue if you give the plotting method a properly formatted string, and the other is that we may want to do this internally to avoid extra work by the user.

You can check out our [contributing guide](https://matplotlib.org/stable/devel/index) for the general contribution workflow. In this particular case, as mentioned above, you may need some familiarity with the TeX internals and/or usetex. You may want to start here: https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/backends/backend_pgf.py (also maybe take a look at https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/text.py or https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/texmanager.py)

**Comment by yjrab** [comment]
Hello @melissawm, if we need to resolve this issue by handling it internally, which function and file should be worked on specifically?

**Comment by melissawm** [comment]
@yjrab I think deciding on that may be part of the work on this issue, so starting from the files I pointed above maybe you can try and see what makes more sense?

**Comment by Vashesh08** [comment]
@melissawm @anntzer 

I think replacing the '%' in text with '\%' will solve the issue. Can you please review the pull request. Thanks

**Comment by sebas-inf** [comment]
Can I pick up this issue as it doesn't seem to be resolved yet and it has been over 3 months since the last PR

**Comment by rcomer** [comment]
Hi @sebas-inf, it looks like that PR is currently waiting on us for a review.  So in this case I wouldn't write a new one.  If you see one that has been waiting on the author for a while, then it would be totally reasonable to open a new one.
